// Copyright 2012 The Chromium Authors
// Use of this source code is governed by a BSD-style license that can be
// found in the LICENSE file.
//
// Sync protocol datatype extension for an extension setting.

// If you change or add any fields in this file, update proto_visitors.h and
// potentially proto_enum_conversions.{h, cc}.

syntax = "proto2";

option java_multiple_files = true;
option java_package = "org.chromium.components.sync.protocol";

option optimize_for = LITE_RUNTIME;

package sync_pb;

// Properties of extension setting sync objects.
message ExtensionSettingSpecifics {
  // Id of the extension the setting is for.
  optional string extension_id = 1;

  // Setting key.
  optional string key = 2;

  // Setting value serialized as JSON.
  optional string value = 3;
}
